airbyte-container-orchestrator/src/test/java/io/airbyte/container_orchestrator/config/ContainerOrchestratorFactoryK8sTest.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.container_orchestrator.config;
+
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+
+import io.airbyte.workers.process.KubeProcessFactory;
+import io.airbyte.workers.process.ProcessFactory;
+import io.micronaut.context.env.Environment;
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import jakarta.inject.Inject;
+import org.junit.jupiter.api.Test;
+
+@MicronautTest(environments = Environment.KUBERNETES)
+class ContainerOrchestratorFactoryK8sTest {
+
+  @Inject
+  ProcessFactory processFactory;
+
+  @Test
+  void processFactory() {
+    assertInstanceOf(KubeProcessFactory.class, processFactory);
```

Comment:
interesting - so this is testing micronaut injects the right class based of env vars?

Reply:
Yeah, the idea is to catch an injection issue during tests. This kind of test won't be necessary if we remove all the `@Factory` classes and make Micronaut manage everything.

Reply:
I kind of like it. We've had several incidents where runtime injection wasn't correct due to either runtime dependencies or configuration. Something like this would have caught it.